parlai/core/torch_agent.py

```diff
@@ -1814,16 +1695,18 @@ def update_params(self):
         # keep track up number of steps, compute warmup factor
         self._number_training_updates += 1
 
+
         # compute warmup adjustment if needed
         if self.opt.get('warmup_updates', -1) > 0:
+            self.scheduler.warmup_step()
             if not hasattr(self, 'warmup_scheduler'):
                 raise RuntimeError('Looks like you forgot to call build_lr_scheduler')
             if self._is_lr_warming_up():
                 self.warmup_scheduler.step(epoch=self._number_training_updates)
```

Comment:
you think we can fold the warmup scheduler into the LRScheduler so that TA makes a very clean call?